### PR TITLE
Improve the Cloud Event Gateway Proxy docs and logs

### PR DESCRIPTION
**Pull request by marcobebway**
**Description**

Changes proposed in this pull request:

- Minor refactoring.
- Add more documentation.
- Log the dispatched event id.

**Related issue(s)**

Fixes https://github.com/kyma-project/kyma/issues/9232.

### CODE REVIEW

**Review by radufa — DISMISSED** [review]

File: components/cloud-event-gateway-proxy/cmd/cloud-event-gateway-proxy/main.go [hunk 1]
```diff
@@ -31,10 +31,7 @@ func main() {
 	defer client.CloseIdleConnections()
 
 	// configure message sender
-	messageSender, err := sender.NewHttpMessageSender(cfg.EmsPublishURL, client)
-	if err != nil {
-		logger.Fatalf("Unable to create message sender with error: %s", err)
-	}
+	messageSender := sender.NewHttpMessageSender(cfg.EmsPublishURL, client)
```
Comment (nachtmaar) [inline components/cloud-event-gateway-proxy/cmd/cloud-event-gateway-proxy/main.go:34]: Can you explain why we are ignoring the error here? 🤔

File: components/cloud-event-gateway-proxy/pkg/handler/handler.go [hunk 2]
```diff
@@ -21,17 +21,21 @@ import (
 )
 
 const (
-	// noDuration signals that the dispatch step hasn't started
+	// noDuration signals that the dispatch step has not started yet.
 	noDuration = -1
 )
 
 var (
+	// additionalHeaders the required headers by EMS for publish requests.
```
Comment (nachtmaar) [inline components/cloud-event-gateway-proxy/pkg/handler/handler.go:29]: I think it reads better like this: What do you think? 

```suggestion
	// additionalHeaders are the required headers by EMS for publish requests.
```

File: components/cloud-event-gateway-proxy/pkg/health/probes.go [hunk 3]
```diff
@@ -9,6 +9,8 @@ const (
 	readinessURI = "/readyz"
 )
 
+// CheckHealth does the health checks for the readiness and liveness probes, otherwise it will let the next
+// handler to take care of the incoming HTTP requests.
```
Comment (nachtmaar) [inline components/cloud-event-gateway-proxy/pkg/health/probes.go:13]:
```suggestion
// handler take care of the incoming HTTP requests.
```

File: components/cloud-event-gateway-proxy/pkg/receiver/receiver.go [hunk 4]
```diff
@@ -11,30 +11,31 @@ import (
 )
 
 const (
-	DefaultShutdownTimeout = time.Minute * 1
+	// defaultShutdownTimeout the default timeout for the receiver to shutdown.
```
Comment (nachtmaar) [inline components/cloud-event-gateway-proxy/pkg/receiver/receiver.go:14]:
```suggestion
	// defaultShutdownTimeout is the default timeout for the receiver to shutdown.
```

File: components/cloud-event-gateway-proxy/cmd/cloud-event-gateway-proxy/main.go (hunk 1 above)
Comment (marcobebway) [reply components/cloud-event-gateway-proxy/cmd/cloud-event-gateway-proxy/main.go:34]: The initial implementation of the `NewHttpMessageSender` function was always returning a `nil` error, which was not needed. So it is simplified to the implementation below:

https://github.com/kyma-project/kyma/blob/0aaf7e9e3e2d520def70de97930645d3cf646a80/components/cloud-event-gateway-proxy/pkg/sender/sender.go#L14-L17

File: components/cloud-event-gateway-proxy/pkg/handler/handler.go (hunk 2 above)
Comment (marcobebway) [reply components/cloud-event-gateway-proxy/pkg/handler/handler.go]: Done.

File: components/cloud-event-gateway-proxy/pkg/receiver/receiver.go (hunk 4 above)
Comment (marcobebway) [reply components/cloud-event-gateway-proxy/pkg/receiver/receiver.go]: Done.

File: components/cloud-event-gateway-proxy/pkg/health/probes.go (hunk 3 above)
Comment (marcobebway) [reply components/cloud-event-gateway-proxy/pkg/health/probes.go]: Done.

File: components/cloud-event-gateway-proxy/cmd/cloud-event-gateway-proxy/main.go (hunk 1 above)
Comment (nachtmaar) [reply components/cloud-event-gateway-proxy/cmd/cloud-event-gateway-proxy/main.go:34]: Ah, I missed the change in the signature.

**Review by nachtmaar — APPROVED** [review]

**Review by radufa — APPROVED** [review]

**Pull request merged**